rmw_connext_shared_cpp/src/wait_set.cpp

```diff
@@ -142,7 +142,7 @@ destroy_wait_set(const char * implementation_identifier, rmw_wait_set_t * wait_s
   RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
     wait_set handle,
     wait_set->implementation_identifier, implementation_identifier,
-    return RMW_RET_ERROR)
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION)
```

Comment:
@ahcorde shouldn't the one above be `RMW_RET_INVALID_ARGUMENT`? Also, consider using the  `RMW_CHECK_ARGUMENT_FOR_NULL` macro.